nomos-core/src/da/attestation/mod.rs

```diff
@@ -1,9 +1,11 @@
 use crate::da::blob::Blob;
 use bytes::Bytes;
+use std::hash::Hash;
 
 pub trait Attestation {
     type Blob: Blob;
+    type Hash: Hash + Eq + Clone;
     fn blob(&self) -> <Self::Blob as Blob>::Hash;
-    fn hash(&self) -> <Self::Blob as Blob>::Hash;
+    fn hash(&self) -> Self::Hash;
     fn as_bytes(&self) -> Bytes;
 }
```

Comment:
Why adding the `Hash` associated type? Reasoning for using the blob one was to mark that the return hash is not the attestation/certificate/whatever but the related blob one. Which now that I think of it it makes sense at type level but maybe not so much as it cannot be enforced (neither I added documentation, completely wrong on my side).

Reply:
> Reasoning for using the blob one was to mark that the return hash is not the attestation/certificate/whatever but the related blob one.

For that we have the `blob` method. Attestations and certificates could be for the same blob but still be different (e.g. produced by different nodes) and we need a way to differentiate those.

> Why adding the Hash associated type?

In theory we could reuse the blob type to enforce that all the items for DA use the same hash type but I could not find a way to write a generic constraint like `Attestation<Blob::Hash = T>` (https://github.com/rust-lang/rust/issues/52662), so I add to add this as a separate trait instead